src/main/java/bio/terra/cli/command/notebook/Create.java

```diff
@@ -0,0 +1,110 @@
+package bio.terra.cli.command.notebook;
+
+import bio.terra.cli.app.AuthenticationManager;
+import bio.terra.cli.app.ToolsManager;
+import bio.terra.cli.model.GlobalContext;
+import bio.terra.cli.model.TerraUser;
+import bio.terra.cli.model.WorkspaceContext;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import picocli.CommandLine;
+
+/** This class corresponds to the third-level "terra notebook create" command. */
+@CommandLine.Command(name = "create", description = "Create a new AI Notebook instance within your workspace.")
+public class Create implements Callable<Integer> {
+
+  @CommandLine.Parameters(
+      index = "0",
+      description =
+          "The unique name to give to the notebook instance. Cannot be changed later. "
+              + "The instance name must be 1 to 63 characters long and contain only lowercase "
+              + "letters, numeric characters, and dashes. The first character must be a lowercase "
+          + "letter and the last character cannot be a dash.")
+  private String instanceName;
+
+  @CommandLine.Option(
+      names = "location",
+      defaultValue = "us-central1-a",
+      description = "The Google Cloud location to create the instance within, e.g. us-central1-a")
+  private String location;
+
+  @CommandLine.Option(
+      names = "machine-type",
+      defaultValue = "n1-standard-4",
+      description = "The Compute Engine machine type of this instance.")
+  private String machineType;
+
+  @CommandLine.Option(
+      names = "vm-image-project",
+      defaultValue = "deeplearning-platform-release",
+      description = "The ID of the Google Cloud project that this VM image belongs to.")
+  private String vmImageProject;
+
+  @CommandLine.Option(
+      names = "vm-image-family",
+      defaultValue = "tf-latest-gpu",
+      description =
+          "Use this VM image family to find the image; the newest image in this family will be used.")
+  private String vmImageFamily;
+
+  // TODO: Add boot disk types & gpu size configs.
+
+  @Override
+  public Integer call() {
+    GlobalContext globalContext = GlobalContext.readFromFile();
+    WorkspaceContext workspaceContext = WorkspaceContext.readFromFile();
+
+    AuthenticationManager authenticationManager =
+        new AuthenticationManager(globalContext, workspaceContext);
+    authenticationManager.loginTerraUser();
+    TerraUser user = globalContext.requireCurrentTerraUser();
```

Comment:
I think it would be good move the bulk of the logic here (~ lines 61-103) into a new class (Notebooks or NotebooksManager maybe?) under the app package. I may be doing some reorganization in that package soon, but regardless I think it's it good to keep the command classes high-level and mostly focused on user input/output.

Reply:
I think I generally agree, but want to push back and figure out where the boundary is. This command is close to a pass-through to a gcloud command. There's a tiny bit of formatting logic (though we could push that back up to the user), but otherwise just grabbing the terra defaults. With another class level, we'd copy all the inputs down a level, but what are we gaining from the abstraction? With something like a remote API, I see it cause there's stuff about setting up the client and formatting request/response. Is it still worthwhile for a delegated CLI call? (I think I'm ok with the answer being, yes, let's do it to have a clear distinction between CLI I/O & execution.)
I think some terminology could also help here. `NotebooksManager` doesn't tell me much. I think we're trying to distinguish between the CLI input/output layer that are these commands and the execution layer.

Reply:
Yes, I'd like to distinguish between the input/output layer and the command logic. Hopefully that will become clearer after [PF-423](https://broadworkbench.atlassian.net/browse/PF-423).

Are you planning to change this from a gcloud command to using a Java client library? If so, I'd be fine punting this code reorg until then.

Reply:
Yes to using a Java client library, but medium term once we have a notebook service up and running. So not for Alpha 1, but yes for Alpha 2. (For the not-yet-created notebooks stop/start commands, those will stay as gcloud commands.) Is that too far off? If yes, could I do it in a follow on to this PR?

Is the correct pattern to create a `NotebooksManager` with a `createNotebook` function with 5 nullable string arguments? (Are defaults command logic or manager logic?)

Reply:
Alpha 2 is fine for me. Main benefit to doing it sooner would be to have all the notebooks related code in one place, but it seems like there's not much shared now anyway.

Yes, I think a Notebooks or NotebooksManager class with a create/Notebook function sounds good. Good question about defaults. I don't have a strong preference, but would probably define them in the notebooks/manager class myself.